kernelci/build.py

```diff
@@ -633,6 +634,24 @@ def build_kernel(build_env, kdir, arch, defconfig=None, jopt=None,
         })
         result = _run_make(target='modules_install', **kwargs)
 
+    # kselftest
+    if result and "kselftest" in defconfig_extras:
+        kselftest_install_path = os.path.join(output_path, '_kselftest_')
+        if os.path.exists(kselftest_install_path):
+            shutil.rmtree(kselftest_install_path)
+        opts.update({
+            'INSTALL_PATH': kselftest_install_path,
+        })
+        #
+        # Ideally this should just be a 'make kselftest-install', but
+        # due to bugs with O= in kselftest Makefile, this has to be
+        # 'make -C tools/testing/selftests install'
```

Comment:
Any chance these bugs can be fixed?  Is there any email thread about this topic to follow what is going?

Reply:
There were some similar discussions, but nothing on this specifially.  I'll start an email thread on the kselftest list.

However, even if this gets fixed upstream, it's unlikely this would be backported, so I think we should stick with the workaround in this patch for awhile, especially if we expect to run kselftest on the stable kernels.

Reply:
FWIW, I *think* this was fixed in v5.7 (see merge 397a97946798890b9bdaa6122fcfad7147690670), but perhaps I'm not understanding the specific bug as it related to running under kernelci's builder.

Reply:
@khilman Can you please add a link to the discussion in the kselftest mailing list archives?

Reply:
There are several related issues around this:
- https://lore.kernel.org/lkml/158351957799.3363.15269768530697526765.stgit@devnote2/
- https://lore.kernel.org/linux-kselftest/MWHPR13MB0895B92C9B4807D94E1E6B04FDE30@MWHPR13MB0895.namprd13.prod.outlook.com/
- https://lore.kernel.org/bpf/cover.1583358715.git.skhan@linuxfoundation.org/

As well as some interesting tools to help sanity check:
- https://lore.kernel.org/linux-kselftest/86508e9d-4aff-d9a0-3783-971a4686ccd1@bird.org/

But again, even though most (maybe all?) of these issues are fixed upstream, it's unlikely these all be backported to stable, so the workaround in this series is still necessary if we want to build kselftest on stable.